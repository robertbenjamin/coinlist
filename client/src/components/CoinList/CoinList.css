.coin-list {
  display: flex;
  flex-wrap: wrap;
}

.coin {
  border: 1px solid #001f3f;
  color: #001f3f;
  padding: 15px;
  margin: 5px;
  flex-grow: 1;
  text-align: center;
}
