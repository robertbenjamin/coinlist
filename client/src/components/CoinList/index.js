export { default } from './CoinList.js';
