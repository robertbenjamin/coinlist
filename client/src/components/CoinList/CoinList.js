import React from 'react';
import './CoinList.css';

const CoinList = props => {
  const coins = props.coins.map(coin => {
    return <div key={coin.id} className="coin">{coin.name} | {coin.symbol}</div>
  });

  return <section className="coin-list">{coins}</section>;
};

export default CoinList;
