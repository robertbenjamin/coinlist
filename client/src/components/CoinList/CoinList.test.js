import React from 'react';
import CoinList from '../CoinList';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const coins = [
    { id: 'ethereum', name: 'Ethereum', symbol: 'ETH' },
    { id: 'bitcoin', name: 'Bitcoin', symbol: 'BTC' }
  ];

  const tree = renderer
    .create(<CoinList coins={coins} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
