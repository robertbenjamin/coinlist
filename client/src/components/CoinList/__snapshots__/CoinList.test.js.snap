// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="coin-list"
>
  <div
    className="coin"
  >
    Ethereum
     | 
    ETH
  </div>
  <div
    className="coin"
  >
    Bitcoin
     | 
    BTC
  </div>
</section>
`;
